### PR TITLE
Ch4 (Zkr), Ch5 (Zkt), Appx B (Zkr rationale) updates, post arch review (for 0.9.4).

### PR DESCRIPTION
**Pull request by mjosaarinen**
Went through the comments we received from arch review and CETG meeting.

* PollEntropy as a pseudoinstruction is gone, now just talks of `sentropy` CSR.
* GetNoise is a name of an optional Custom CSR defined at the end of Appendix B.
* Generally shuffled things around so that Ch4 has a minimum amount of non-ISA stuff.
* Added a testing checklist at the beginning of Appendix B.
* Removed the RV128 instructions from Ch5, added an explanation on need for `sentropy` to have secret-independent latency.
* The pull also touches on the bibliography (adding a couple of things, and removes the leveloffset for Ch5.
* There are two state diagrams; the main one without the test interface, another one in the appendix with it. Now includes a Makefile for making svg from tikz figures (and their sources).
* Various other edits.

### CODE REVIEW

File: doc/scalar/riscv-crypto-scalar-appx-entropy-source.adoc [hunk 1]
```diff
@@ -123,7 +194,7 @@ the risk of state compromise.
 A blocking instruction may be easier to use, but most users should
 be querying a (D)RBG instead of an entropy source.
 Without a polling-style mechanism, the entropy source could hang for
-thousands of cycles under some circumstances. The `wfi`
+thousands of cycles under some circumstances. A `wfi` ot `pause`
```
Comment (ben-marshall) [inline doc/scalar/riscv-crypto-scalar-appx-entropy-source.adoc:197]: ot -> or

File: doc/scalar/riscv-crypto-scalar-entropy-source.adoc [hunk 2]
```diff
@@ -106,128 +89,73 @@ false positive in the continuous testing procedures. In case of a fatal
...
-larger than 2^-192^ of occurring (min-entropy 192 bits).
+The main requirement is that 2-to-1 cryptographic post-processing
+in 256-bit input blocks will yield 128-bit "full entropy" output blocks.
+Users of `sentropy` may make this conservative assumption but are not
+prohibited from using more than twice the number of seed bits relative
+to the desired resulting entropy.
 
-Depending on the implementation and process instance, the source can be one of
-three things:
+The entropy source can meet one (or more) of three sets of requirements:
```
Comment (ben-marshall) [inline doc/scalar/riscv-crypto-scalar-entropy-source.adoc:117]: Aren't these requirements what the arch-review folks wanted in the appendix?

In any case, saying the ES "can meet one (or more) of three sets of requirements" doesn't actually convey a requirement I can test, only a possibility.

**Comment by ben-marshall** [comment]
All looks fine to me. I'll leave the PR open until weds for others to look at.

File: doc/scalar/riscv-crypto-scalar-entropy-source.adoc (hunk 2 above)
Comment (mjosaarinen) [reply doc/scalar/riscv-crypto-scalar-entropy-source.adoc:124]: Well, the idea is that entropy itself is not within the scope of architectural tests, just its behavior in relation to the interface (beginning of Appendix). If the interface is implemented correctly, the back-end entropy source module can be changed. In other words, the "entropy source interface" is within scope, but the "entropy source" is not, right? I hope that Section B.1. "A Design and Self-Certification Checklist" clarifies this a bit.

**Comment by mjosaarinen** [comment]
After discussions with the arch reviewers, now additionally:
* Includes the preliminary CSR assignment 0x546.
* Discusses destructive and non-destructive writes, `csrrw rd, sentropy, x0`.
* Trap in case a read is attempted without the side-effect of clearing the source.
* Minor non-funcitonal edits and improvements all around.

**Comment by ben-marshall** [comment]
Is this ready to merge then?

**Comment by mjosaarinen** [comment]
Yep sure go ahead!

**Pull request merged**